### DISCUSSION TITLE
Setting certain time interval when times of day can be changed by user

### DISCUSSION THREAD

**Discussion by jtk2ho** [KL-Psychological-Methodology/ESMira#19] [Ideas]
Hello,

I was wondering if it would be possible to create a setting under triggers. Currently, "times of day can be changed by user" may be selected but this allows users to set any schedule they choose during the day. It could be helpful to allow times to be changed by user but within a set time interval (e.g., in the study I am running, the interval must last 12 hours. It would be useful to have a way to allow participants to choose 7am - 7pm vs. 9am - 9pm etc without allowing them to select a time period less than 12 hours).

Thank you!

**Comment by SelinaDev** [comment]
Hi, and thank you for your comment. I have some more questions about your idea, just to make sure I unterstand it. Do you mean that you want to coordinate multiple questionnaires or signals? E.g., have two signals separated by a fixed interval (12 hours in your example), but have the user select the start of that interval (from within a set time frame, 7am to 9 am in your example)?
I have a package of changes to how signals/alarms should work that is part of my backlog. If I understood your proposal correctly, it sounds like an extension to something I wanted to work on as part of that anyway. I can't say when I will get to all thet, as there are other things on my backlog that I likely will work on before that, but I will keep this in mind.
Best, 
Selina

**Comment by jtk2ho** [comment]
Hi Selina, apologies if my comment wasn't clear! I was thinking of something like this: a study may want to notify participants randomly 5 times within a 12 hour period with a repeated questionnaire. Because of the nature of the study, it may be important for a full 12 hour range to be covered. In my case, I would like these questionnaires to be at least 1 hour apart (and ESMira does currently have a setting to set minimum time between questionnaires, which is really useful). However, if participants are being allowed to determine their own schedules, there is currently does not seem to be any way to make participants adhere to a 12 hour timeframe, as participants can freely set both start and end times in the schedule. Hope that helps clarify, thanks!

**Comment by JodliDev** [comment]
The trickiest thing for that would be to find space in the admin interface (since there are already many possible options per time of day in one line) :D
I will open an issue. I can not promise a timeline but I might implement it in the near future (or maybe SelinaDev)

**Reply by jtk2ho** [reply]
Absolutely, there's no rush at all but thank you for taking it into consideration!
